save-demo/src/main/kotlin/com/saveourtool/save/demo/service/AbstractDemoService.kt

```diff
@@ -1,21 +1,21 @@
 package com.saveourtool.save.demo.service
 
-import com.saveourtool.save.demo.diktat.DiktatAdditionalParams
-import com.saveourtool.save.demo.diktat.DiktatDemoResult
+import com.saveourtool.save.demo.DemoResult
+import com.saveourtool.save.demo.DemoRunRequest
 import com.saveourtool.save.demo.runners.Runner
 
 /**
  * Abstract service interface for different demonstration services
  */
-abstract class AbstractDemoService<in P : DiktatAdditionalParams, in K : Any, out R : DiktatDemoResult>(
-    private val runner: Runner<P, K, R>,
+abstract class AbstractDemoService(
+    private val runner: Runner,
 ) {
     /**
-     * Run demo on [demoFileLines] with [diktatAdditionalParams] and return result as [DiktatDemoResult]
+     * Run demo on [demoFileLines] with [runRequest] and return result as [DemoResult]
      *
      * @param demoFileLines list of lines of input file that will be used for demo
-     * @param diktatAdditionalParams additional params as [DiktatAdditionalParams]
-     * @return report as [DiktatDemoResult]
+     * @param runRequest additional params as [DemoRunRequest]
+     * @return report as [DemoResult]
      */
-    abstract fun launch(demoFileLines: List<String>, diktatAdditionalParams: P? = null): R
+    abstract fun launch(demoFileLines: List<String>, runRequest: DemoRunRequest = DemoRunRequest.empty): DemoResult
```

Comment:
do we need to pass `demoFileLines` twice?

Reply:
Done